### ISSUE TITLE
can't able to take mongodb DUMP getting autherror and not authorized for query on my-database.system.indexes Err

### ISSUE THREAD

**Issue by SHUFIL** [bitnami/charts#3026]
**Which chart**:
The name (and version) of the affected chart
CHART mongodb-7.14.8, APP VERSION-4.2.8

**Describe the bug**
when i take mongodb dump its getting error , err: "not authorized for query on mydb.system.indexes"
If we use "-u root" getting authentification error  
**To Reproduce**
1. install bitnami/mongodb - helm install --name mongodb --set auth.rootPassword=secretpassword,auth.username=my-user,auth.password=my-password,auth.database=my-database bitnami/mongodb
or clone and used -f values.yaml
 
then export root password  "export MONGODB_ROOT_PASSWORD=$(kubectl get secret --namespace default mongodb -o jsonpath="{.data.mongodb-root-password}" | base64 --decode)"
Now I can login without any issue with 
   kubectl run --namespace default mongodb-client --rm --tty -i --restart='Never' --image docker.io/bitnami/mongodb:4.2.8-debian-10-r7 --command -- mongo admin --host mongodb --authenticationDatabase admin -u root -p $MONGODB_ROOT_PASSWORD
 
Problem only when take backup
2. create a DB name my-database, and add some data - db.myCollection.insertOne ( { key1: "value1" });
3. created a pvc , i have used below yaml
    apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: mongoutil-mongodump
  namespace: kube-system
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
4. create and run  kubernetes Job 
    apiVersion: batch/v1
kind: Job
metadata:
  name: mongoutil-mongodump
  namespace: kube-system
spec:
  template:
    spec:
      containers:
      - name: mongoutil-mongodump
        image: bitnami/mongodb
        command: ["mongodump", "-h", "mongodb", "--db", "my-database", "--out", "/dump", "--authenticationDatabase", "admin", "-p", "my-password", "--verbose"]
        volumeMounts:
        - mountPath: "/dump"
          name: mongodump
      volumes:
      - name: mongodump
        persistentVolumeClaim:
          claimName: mongoutil-mongodump
      restartPolicy: Never
  backoffLimit: 4
5. getting below error
   MongoDB shell version v4.2.8
connecting to: mongodb://mongodb:27017/admin?authSource=admin&compressors=disabled&gssapiServiceName=mongodb
2020-07-05T14:37:41.489+0000 E  QUERY    [js] Error: Authentication failed. :
connect@src/mongo/shell/mongo.js:341:17
@(connect):2:6
2020-07-05T14:37:41.490+0000 F  -        [main] exception: connect failed
2020-07-05T14:37:41.490+0000 E  -        [main] exiting with code 1

Then i have create one POD  , name helloworld-deployment it is  attached volume and installed mongodb-client then run 
kubectl exec helloworld-deployment-c7599775b-dz5n5 -it -- mongodump -h mongodb --db my-database --out  /data --authenticationDatabase admin -p $MONGODB_ROOT_PASSWORD --verbose

This time getting error is - 
  Fri Jul  3 03:19:13.471 creating new connection to:mongodb:27017
Fri Jul  3 03:19:13.473 BackgroundJob starting: ConnectBG
Fri Jul  3 03:19:13.473 connected connection!
connected to: mongodb
Fri Jul  3 03:19:13.474 DATABASE: my-database       to     /data/my-database
Fri Jul  3 03:19:13.475 User Assertion: 13106:nextSafe(): { $err: "not authorized for query on my-database.system.indexes", code: 13, ok: 0.0 }
assertion: 13106 nextSafe(): { $err: "not authorized for query on my-database.system.indexes", code: 13, ok: 0.0 }
command terminated with exit code 255
 
     If we use "-u root or any oter user, then getting authentification error"

Tryied to resolve i have installed using below command
 helm install --name mongodb --set mongodbRootPassword=password123,mongodbUsername=testuser,mongodbPassword=password123,mongodbDatabase=my-database -f values.yaml path-to-path
But getting same error

**Version of Helm and Kubernetes**:

- Output of `helm version`: 

```
(paste your output here)
```Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}


- Output of `kubectl version`:

```Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T11:48:36Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

(paste your output here)
```minikube version: v1.10.1


**Additional context**
Add any other context about the problem here.

**Comment by javsalgar** [comment]
Hi,

I was unable to reproduce the issue. I deployed the chart with the parameters you provided, added some collections and was able to dump


```
mongodump -u root -h 127.0.0.1 --db my-database --out /tmp/data --authenticationDatabase admin -p $MONGODB_ROOT_PASSWORD --verbose
Handling connection for 27017
Handling connection for 27017
2020-07-08T11:07:23.215+0200	dumping up to 3 collections in parallel
Handling connection for 27017
Handling connection for 27017
2020-07-08T11:07:23.217+0200	writing my-database.contacts to /tmp/data/my-database/contacts.bson
2020-07-08T11:07:23.220+0200	done dumping my-database.contacts (0 documents)
2020-07-08T11:07:23.242+0200	writing my-database.myColl to /tmp/data/my-database/myColl.bson
2020-07-08T11:07:23.242+0200	writing my-database.log to /tmp/data/my-database/log.bson
2020-07-08T11:07:23.245+0200	done dumping my-database.myColl (3 documents)
2020-07-08T11:07:23.245+0200	done dumping my-database.log (0 documents)
```

In my case I used kubectl port-forward and a local mongodb client

**Comment by SHUFIL** [comment]
then what i done wrong , i still getting same issue , yes i used port-forward  , and tryed from local system, getting below error
```
ubuntu@server5:~$ mongo --host 127.0.0.1 --authenticationDatabase admin -p                                                                                              MongoDB shell version: 2.6.10
Enter password:
connecting to: 127.0.0.1:27017/test
> show dbs
2020-07-08T16:57:37.826+0530 listDatabases failed:{
        "ok" : 0,
        "errmsg" : "command listDatabases requires authentication",
        "code" : 13,
        "codeName" : "Unauthorized"
} at src/mongo/shell/mongo.js:47
```

**Comment by javsalgar** [comment]
The difference in your command is that you do not use the `root` user, could that be the reason?

**Comment by SHUFIL** [comment]
I have used helm 3, and used `helm install mongodb bitnami/mongodb`, now i can able login inside the POD 
`mongo admin --host "mongodb" --authenticationDatabase admin -u root -p $MONGODB_ROOT_PASSWORD`

helm3 no need to install helm tiller, also no need to use --name <helm-name>

**Comment by javsalgar** [comment]
Thanks for letting us know! Do not hesitate to open a new ticket if you find any other issue
